motoko/invoice-canister/examples/motoko-seller-client/backend/src/seller/main.mo

```diff
@@ -0,0 +1,118 @@
+import Invoice "canister:invoice";
+
+import Hash       "mo:base/Hash";
+import HashMap    "mo:base/HashMap";
+import Iter       "mo:base/Iter";
+import Nat        "mo:base/Nat";
+import Principal  "mo:base/Principal";
+import Result     "mo:base/Result";
+import Text       "mo:base/Text";
+
+actor Seller {
+
+  let ONE_ICP_IN_E8S = 100_000_000;
+
+  stable var invoicesStable : [(Nat, Invoice.Invoice)] = [];
+  var invoices: HashMap.HashMap<Nat, Invoice.Invoice> = HashMap.HashMap(16, Nat.equal, Hash.hash);
+
+  stable var licensesStable : [(Principal, Bool)] = [];
+  var licenses: HashMap.HashMap<Principal, Bool> = HashMap.HashMap(16, Principal.equal, Principal.hash);
+
+// #region create_invoice
+  public shared ({caller}) func create_invoice() : async Invoice.CreateInvoiceResult {
+    let invoiceCreateArgs : Invoice.CreateInvoiceArgs = {
```

Comment:
Access control?

Reply:
I'll create a follow-up issue for access control